Replaced service borg dice with new borg die.

Faptastic opened the pull request:
8 dice squashed into one, the bloat is gone!
I would welcome an improved underlay if you have one, not my forte.

Exxion commented:
Now a d00 can't roll below 10
There's a reason it couldn't reach 100 before
Because it goes from 0-90
Which you add to a d10 to get a 1-100 roll

PJB3005 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -7,12 +7,40 @@
+	set name = "Set Poly-die Sides"
+	set category = "Object"
+	set src in range(0)
+	set_sides()
```
Gonna need some sanity (is the person alive, etc...)

ihadtoregisterforthis commented:
Maybe a digital die would be more fitting than one made from a "mysterious material".

ComicIronic commented:
Digi-die does sound like a good idea.
Also, perhaps get a new icon.

Faptastic commented:
![digidie](https://cloud.githubusercontent.com/assets/14299601/16616968/adc74614-4378-11e6-8b28-678924e0d5f7.jpg)
This better? I'm not a sprite artist but its hopefully decent enough to look like a small tablet with a picture of a die on it. If someone has a better one I welcome it.

At least this way you can keep the fancy dice roll icons but handwave the fact the die magically changes shape,

PJB3005 commented:
Could you add the change functionality to the dice as am alt click functionality? That'd be a lot quicker than right click.

ComicIronic commented:
Wait, it's a verb?

DELET

PJB3005 commented:
Issue with removing the verb is that now a lot of people won't be able to figure out how to change the amount of sides.

ComicIronic commented:
It could be a contextual click on the icon which had some design to encourage you to click it.

Faptastic commented:
I've added a contextual click and a little yellow button to go with it, seems to work well enough.
Also tried the device as a human should that ever happen, works exactly like other die.
The description was also modified to mention the button as well as the changelog, I'm sure the player will eventually get the hint.

Unless you have some tweaks to this or suggestions I'm about done.

ComicIronic commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -290,3 +293,54 @@
+		return 1
+
+/obj/item/weapon/dice/borg/proc/set_sides()
+	var/S = input("Number of sides:") as null|anything in possible_sides
```
This is going to be a sanity issue. Pass in user for this proc and check the user is still next to the dice after the input.

PJB3005 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -326,11 +326,13 @@
 		return 1
 
-/obj/item/weapon/dice/borg/proc/set_sides()
+/obj/item/weapon/dice/borg/proc/set_sides(mob/usr)
```
Uh, I'm wondering how this even compiled but make this `user` instead and then use it below.

Faptastic replied:
Do I change the other instances of 'usr' as well that don't cause a compile error when changed? I remember some sort of crusade against either usr or user but I couldn't find which.
I'm testing each revision before it's pushed so thats something at least.

ComicIronic replied:
usr is a special Byond var, and you should almost never be using it. If you're defining your own var, use user.

ComicIronic commented:
Nicely done on the context clicker, people need to use those more (and not just because I made them)
